extensions/applicationinsights-cfgsync-js/src/CfgSyncHelperFuncs.ts

```diff
@@ -39,3 +42,114 @@ export function replaceByNonOverrideCfg<T=IConfiguration & IConfig, T1=NonOverri
     // if errors happen, do nothing
     return cfg;
 }
+
+
+//                                                     CDN Mode, value = B (CDN value = B)
+//                                |--------------------------------------------------------------------------|
+//                                |                    | none        | disabled    | enabled     | force     |
+//                                | ------------------ | ----------- | ----------- | ----------- | --------- |
+// | User Mode, value = A || C    | none               | none        | disabled    | disabled    | enabled   |
+// (user Value = A)               | disabled           | disabled    | disabled    | disabled    | enabled   |
+// (SDK Defaults = C)             | enabled            | enabled     | disabled    | enabled     | enabled   |
+//                                | none(blockCdn)     | none        | none        | none        | none      |
+//                                | disabled(blockCdn) | disabled    | disabled    | disabled    | disabled  |
+//                                | enabled(blockCdn)  | enabled     | enabled     | enabled     | enabled   |
+
+//                                cdn Mode (cdn Value = B)
```

Comment:
I went back to the first instance of the table to make sure that we didn't miss anything

```
// This matrix identifies how feature based overrides are selected (if present),
// feature overrides are defined as a map of encoded keys with the value like
// featureOptIn: {
//     "disableUnloadEvent" : {
//         cfg: {
//            "disablePageUnloadEvents": [ "unload" ]
//         }
//     }
// }
//
// ---------------------------------------------------------------------------------
//                                cdn Mode (cdn override value = B)
//                    |------------------------------------------------------------|
//                    | Value mapping      | none    | disabled | enabled | force  |
//                    | ------------------ | --------| ---------| --------| ------ |
// | User Mode        | none               | A       | A        | A || B  | B || A |
// (user or SDK       | disabled           | A       | A        | A       | B || A |
// default Value = A) | enabled            | A       | A        | A || B  | B || A |
//                    | none(blockCdn)     | A       | A        | A       | A      |
//                    | disabled(blockCdn) | A       | A        | A       | A      |
//                    | enabled(blockCdn)  | A       | A        | A       | A      |
//
// Where the "value" is an object (map) of encoded key/values which will be used to override the real configuration
// A = either the user supplied value (via the `config.featureOptIn[name]`) or if not defined by the user the SDK defaults of the same.
// B = The value obtained from the CDN for the named feature
// These are evaluated based on the above matrix to either
// - A (Always the user/sdk value)
// - B (Always the value from the CDN)
// - A || B (If A is null or undefined use the value from the CDN (B) otherwise A)
// - B || A (If B is null or undefined use the user/SDK value otherwise use the value from the CDN)
//
// The result of the value may also be null / undefined, which means there are no overrides to apply when the feature is enabled
```

If we allow the programmatic "enable/disable" feature to provide some additional configuration (ie. for `B` overrides) then we may have to revisit this again. "But" as they will be able to "change" the `A` value in the configuration themselves (including the `featureOptIn` config we can probably avoid it (at least for now) -- although it might make testing easier and maybe even the code...

Reply:
Went to add some additional comments about the `disabled` state returning `A`, if we rephrase this matrix as the "overrides" that will be applied to when the feature is "enabled" then for the `disabled` states should this return anything???

Additional thought, should we have 2 sets of overrides available in the `featureOptIn` so that when enabled it applies the "enabled" set and when disabled it applies the "disable" set??? With the `none` state just not applying any overrides....
Then it should always be
- `A[enabled] || B[enabled]`
- `A[disabled] || B[disabled]`
- null

And for the force options
- `B[enabled] || A[enabled]`
- `B[disabled] || A[disabled]`

Reply:
`enableCfgVal`, `disableCfgVal` inside map inside `featureOptIn`